internal/clients/aws.go

```diff
@@ -111,6 +113,37 @@ func (m *metaOnlyPrimary) Meta() any {
 	return m.meta
 }
 
+// withExternalAPICallCounter configures an AWS SDK v2 stack (client)
+// with an API call counter.
+func withExternalAPICallCounter(stack *middleware.Stack) error {
+	externalAPICallCounterMiddleware := middleware.DeserializeMiddlewareFunc("externalAPICallCounter",
+		func(ctx context.Context, input middleware.DeserializeInput, next middleware.DeserializeHandler) (middleware.DeserializeOutput, middleware.Metadata, error) {
+			serviceID := awsmiddleware.GetServiceID(ctx)
```

Comment:
Did we handle the inconsistent service names between the AWS SDK v1 & v2? I feel it would be less confusing if we can emit the same name regardless of the version of the SDK the resource uses.

Reply:
We did. Inconsistency in naming was due to the fact that SDK v1 had `ServiceName`, which we used, in addition to `ServiceID`, which we should have used. Now that both SDK v1 and v2 use `ServiceID`, they all count towards the same metric.